src/cargo/ops/common_for_install_and_uninstall.rs

```diff
@@ -546,9 +546,29 @@ impl InstallInfo {
 
 /// Determines the root directory where installation is done.
 pub fn resolve_root(flag: Option<&str>, gctx: &GlobalContext) -> CargoResult<Filesystem> {
-    let config_root = gctx
-        .get::<Option<ConfigRelativePath>>("install.root")?
-        .map(|p| p.resolve_program(gctx));
+    let config_root = match gctx.get::<Option<ConfigRelativePath>>("install.root")? {
+        Some(p) => {
+            let resolved = p.resolve_program(gctx);
+            if resolved.is_relative() {
+                let definition = p.value().definition.clone();
+                if matches!(definition, Definition::Path(_)) {
+                    let suggested = format!("{}/", resolved.display());
+                    gctx.shell().warn(format!(
```

Comment:
I wonder of we what to use annotate-snippets style report the, as this kinda has three sentence, so probably something like:


```
warn: the `install.root` value `t1` defined in [ROOT]/.cargo/config.toml without a trailing slash is deprecated
  | 
  = note: a future version of Cargo will treat it as relative to the configuration directory
  = help: add a trailing slash (`{}`) to adopt the correct behavior and silence this warning
  = note: see more at https://doc.rust-lang.org/cargo/reference/config.html#config-relative-paths"
```

Reply:
Beautiful!

Updated.